### ISSUE TITLE
Native protection settings for interactions with restrict or remove own-avatar agency, cause discomfort, etc

### ISSUE THREAD

**Issue by gentlecolts** [Yellow-Dog-Man/Resonite-Issues#337]
### Is your feature request related to a problem? Please describe.

Many common issues I've experienced that emerge from this platform's highly-interactive nature should provide options for opting out.  Some examples include:

- anchoring
- knockback
- personal space
- posession of proxies
- writing to user position, rotation, scale
- screen space effects
- forcably changing avatar
- nonconsentual reparenting

### Describe the solution you'd like

While these interactions are an emergent consequence of the fundamental design of this platform, and in many cases are part of the fun for a lot of people, users should have a means of opting out of these kinds of interactions.  This opt-out should come in the form of native comfort settings which cannot be altered by in-world entities.  It'd also be nice if we could, additionally, allow trusted users from being except from our personal protections, disable on a per-session basis, etc

### Describe alternatives you've considered

Some of these can be mitigated by in-game protection systems, however these systems have several issues:

1. A user has to discover them, it's not the "native experience"
2. They are limited in what they can do, for example, no system can protect users from having to their transform written to without using hacky, unsuported means
3. in-game systems can be circumvented, which can lead to arms-races that are unhealthy for the community at large

### Additional Context

zangooseoo

**Comment by gentlecolts** [comment]
Some of these effects are also important for worlds, for example.  It might be nice if worlds can have a prompt stating what permissions they want, (ex, an amusement park world may want to anchor users, a game world might want to apply knockback), allowing easy acquisition of these permissions where necessary

**Comment by MargoSizemore** [comment]
I would really appreciate this

**Comment by JackTheFoxOtter** [comment]
I don't see how this could be disabled on an inidvidual level, but it could definitely be disabled on a world / session level, which I think is also covered by the plans for hard permissions. An option to disable this on the client side could lead to a lot of world breakage, but it would be nice if this can be at least communicated to the user before they join the world, in an abstract way kind of analog to how apps tell you what access they need before installing them. "In this world, systems can change your avatar" for example.

**Comment by gentlecolts** [comment]
A lot of these are very "wishlist" items, things that I'd like to see in a perfect system.  I definitely believe there's incremental steps that can be taken along the way, and this doesn't have to be implemented all at once.  For example, having native protections built into the client as a start, then over time giving options for making it more granular would be, in my opinion, a good approach.  There's a very "right now" problem of user comfort that needs to be addressed, imo, years ago, and fixing that problem should be a major priority.

**Comment by gentlecolts** [comment]
I'm also not entirely concerned about world breakage.  Ideally a system built around requesting permissions should emerge, but I personally see the current standard of free, no-consent interaction to be a bug, not a feature

**Comment by JackTheFoxOtter** [comment]
I agree that this is an issue that needs addressing sooner rather than later. That's why I don't think a small, incremental fix is the way to go - instead, effort should be put into providing a robust implementation for hard permissions that allows session hosts much more granular fine control over what happens in their session, and transparently communicates to users what they should expect before joining the session.

**Comment by epicEaston197** [comment]
Can we please get a official issue on the hard permission system if you wouldn't mind
